### PR TITLE
Feature/build and start process

### PR DESCRIPTION
**Pull request by Villiamfj**
Things done:
- Implemented build from git, where the client is expected to send a string containing either:
   - link to repo
   - "none" for just observing
- Disconnecting clients after repo has been build and executed (and all measurements are send)
   - This is an indication to the client that the measurement is done.
- Disconnecting clients that cant accept data
- Some error handling

### CODE REVIEW

File: crates/server/src/listener.rs [hunk 1]
```diff
@@ -105,14 +110,56 @@ fn handle_local_connection(mut socket: tokio::net::TcpStream) {
...
 
-fn handle_remote_connection(
+async fn handle_remote_connection(
     remote_tcpstreams: Arc<Mutex<Vec<std::net::TcpStream>>>,
-    socket: tokio::net::TcpStream,
+    mut socket: tokio::net::TcpStream,
 ) {
+    // Getting repo from client
+    let mut buf = Vec::new();
+    socket.read_buf(&mut buf).await.unwrap(); // url is expected to be within on packet
```
Comment (Jakobzs) [inline crates/server/src/listener.rs:119]: Mener den teknisk set kan læse 0 her fordi Vec::new() er et vektor af en størrelse af 0

File: crates/server/src/listener.rs [hunk 2]
```diff
@@ -105,14 +110,56 @@ fn handle_local_connection(mut socket: tokio::net::TcpStream) {
...
+        let res = GitBuild {}.build(repo);
+        match res {
+            Ok(_) => {}
+            Err(e) => {
+                println!("Failed to build and run repo: {:?}", e);
+            }
+        }
+
+        // waiting for measurements to be sent
+        while !LOCAL_CLIENT_PACKET_QUEUE.is_empty() {
```
Comment (Jakobzs) [inline crates/server/src/listener.rs:148]: Går ud fra at man venter indtil programmet er lukket, derfor venter indtil queuen er tom?

File: crates/server/src/listener.rs [hunk 3]
```diff
@@ -105,14 +110,56 @@ fn handle_local_connection(mut socket: tokio::net::TcpStream) {
...
+            thread::sleep(Duration::from_secs(1));
+        }
+
+        // Disconnecting client measurement is done
+        // TODO this can break with multiple clients are connected.
```
Comment (Jakobzs) [inline crates/server/src/listener.rs:153]: Jep tænker det kan være et problem

File: Client.js [hunk 4]
```diff
@@ -6,19 +6,22 @@ const Net = require('net');
 // The port number and hostname of the server.
 const port = 6969;
 const host = "127.0.0.1";
-const end = "end";
+const endString = "end";
+const end = new Buffer.from(endString);
+const repo = "https://github.com/cs-24-pt-10-01/HotspotBenchmarkJS.git";
```
[the remark this answers is not in the document]
Comment (Villiamfj) [reply Client.js:11]: ja, det kan sættes ind om argumenter

**Review by Jakobzs — APPROVED** [review]

**Pull request merged**